### PR TITLE
LPFG-929: Add a new organisation

### PR DESCRIPTION
**Pull request by m24murray**
- adding new organisation
- typeahead logic
- validation for adding org
- view organisation page will be coming in next PR, so currently it will just redirect to org list

_Also includes closed PR: https://github.com/Civil-Service-Human-Resources/lpg-management/pull/128_
- controller and template for manage organisations screen
- i've updated our prettier config (extending line length) so there's quite a few formatting changes

### CODE REVIEW

File: src/controllers/organisation/model/organisationalUnitFactory.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	}
+
+	public create(data: any): any {
+		const organisationalUnit: OrganisationalUnit = new OrganisationalUnit()
+
+		organisationalUnit.id = data.id
+		organisationalUnit.name = data.name
+		organisationalUnit.code = data.code
+		organisationalUnit.paymentMethods = data.paymentMethods
+		organisationalUnit.subOrgs = data.subOrgs
```
Comment (Peter-D-Taylor) [inline src/controllers/organisation/model/organisationalUnitFactory.ts]: Shouldn't the sub organisationalUnits be passed to create as well
`organisationalUnit.subOrgs = (data.subOrgs || []).map(this.create)`

File: views/page/organisation/add-organisation.html [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+{% block pageTitle %}{{ i18n.organisations.title.addANewOrganisation }} - {{ i18n.proposition_name }}{% endblock %}
+
+{% set banner = true %}
+{% set backButton = "/content-management/organisations" %}
+
+{% block content %}
+{% set form = sessionFlash.form %}
+
+    {{ errorSummary(errors.fields) }}
```
Comment (Peter-D-Taylor) [inline views/page/organisation/add-organisation.html]: Looks like there's a duplicate errorSummary here

File: src/controllers/organisation/model/organisationalUnit.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import {IsNotEmpty} from 'class-validator'
```
Comment (robleney-kainos) [inline src/controllers/organisation/model/organisationalUnit.ts]: Is there a reason this isn't under `/src/csrs/model`?

Comment (m24murray) [reply src/controllers/organisation/model/organisationalUnit.ts]: Good catch, that's it moved to csrs now

**Pull request merged**